spec/release-lock-file.schema.json

```diff
@@ -0,0 +1,134 @@
+    },
+    "sources": {
+      "title": "Sources",
+      "type": "array",
+      "description": "The sources field defines a set of source files that comprise the source code for the project. This list should be included in all manifests. Package managers should use this field to inform population of the sources field in the release lock file.\n\n * All keys **must** conform to *one of* the following formats.\n     * Begins with a `./` to denote that it is a filesystem path.\n     * Is a valid contract name matching the regex `[_a-zA-Z][_a-zA-Z0-9]*`.\n * All keys which are formatted to be filesystem paths **must** conform to *all* of the following rules:\n     * Resolving the filesystem paths must result path that is *under* the current working directory.\n * *If* the key is a contract name the value **must** be the source string for that contract.\n * *If* the key is a filesystem path the value **must** conform to *one of* the following formats.\n     * IPFS URI\n         * *If* the resulting document is a directory the key should be interpreted as a directory path.\n         * *If* the resulting document is a file the key should be interpreted as a file path.\n",
+      "items": {
+        "title": "Source Path",
+        "type": "string",
+        "description": ": Strings must be formatted as valid filesystem paths. All paths must be relative paths beginning with ./. All paths must resolve to to a path located within the base directory of the package (i.e., the location of epm.json"
+      },
+      "default": []
+    },
+    "chain": {
+      "title": "Chain",
+      "type": "array",
+      "description": "The `chain` field defines the blockchain that should be used for any addresses provided with this package.  A blockchain is defined using [BIP-122](https://github.com/bitcoin/bips/blob/master/bip-0122.mediawiki).  A matching blockchain is one on which all resources defined by the list of BIP122 URIs can be found. *If* this release lock file includes any addressed contracts this field **must** be present.  Convention is to define a chain using a single URI which points to the latest observable block hash and includes the genesis hash as the `chain_id`.",
+      "default": [],
+      "items": {
+        "title": "URI",
+        "description": "must be valid BIP122 URIS",
+        "type": "string"
+      }
+    },
+    "contracts": {
+      "title": "Contracts",
+      "type": "object",
+      "description": "The `contracts` field declares information about the deployed contracts included within this release.",
+      "patternProperties": {
+        "^[a-zA-Z][-a-zA-Z0-9_]*$": {
+          "type": "object",
+          "description": "A *Contract Instance Object* is an object with the following key/values.",
+          "properties": {
+            "contract_name": {
+              "type": "string",
+              "description": "Valid contract name matching regular expression `[_a-zA-Z][_a-zA-Z0-9]*]`"
```

Comment:
Can we add `"pattern": "^[a-zA-Z][-a-zA-Z0-9_]*$"` to this to enforce the contract naming scheme.

http://json-schema.org/latest/json-schema-validation.html#rfc.section.5.8